dask/sizeof.py

```diff
@@ -47,6 +48,14 @@ def sizeof_python_dict(d):
     )
 
 
+@sizeof.register(chain)
```

Comment:
I'm not sure if this belongs here, or in the `total_mem_usage` method in `bag/core.py`. 

But the point of adding this was to make sure that repartitioning a bag multiple times in a row work...on subsequent repartitions, all the partitions were an `itertools.chain` object, and `dask.sizeof.sizeof` didn't have a way to accurately gauge memory usage of that.

Reply:
hmm. That is an interesting question. This seems like the right place to me. The point of the `sizeof` functions is to keep track of how many bytes the system would need to keep the object in memory.

Reply:
We'll want to make sure that this isn't burning a consumable, and that this data will still be around for future use.  Probably before we repartition we'll want to reify all iterators into lists anyway.

Reply:
yes good point about not consuming the iterator. I decided to try just making a copy of the `chain` object to preserve the original, and using this same calculation.